plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -74,6 +89,14 @@ public TableType getTableType()
         return tableType;
     }
 
+    @JsonProperty("tableSchema")
+    @JsonSerialize(using = SchemaSerializer.class)
```

Comment:
This won't work, unfortunately, since only `com.fasterxml.jackson.annotation.` is part of the SPI. The databind annotations and referenced classes will not be visible to the engine.

Reply:
Probably the easiest way is to create a wrapper object like `SerializableSchema` that uses `@JsonValue` and `@JsonCreator`.